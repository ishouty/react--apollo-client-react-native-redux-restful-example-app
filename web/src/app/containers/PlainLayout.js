import React, { Component } from "react"
import {Banner} from "../components/Common/Common"

export default class PlainLayout extends Component {

    constructor (props, context) {
        super(props, context)

    }

    render () {

        return (

            <div id="layout">

                <div id="main">
                    <Banner />
                    {/* this will render the child routes */}
                    {this.props.children}

                </div>

            </div>

        )
    }
}
