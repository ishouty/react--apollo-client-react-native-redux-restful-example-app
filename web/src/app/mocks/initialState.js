import { NavigationLinkText } from '../constants/text/text'

export const initialState = {
  users: {
    users: [],
    page: 0,
    totalPages: 20
  },
  loading: {
    loading: {
      common: false
    }
  },
  notifications: {
    banner: {
      fail: false,
      success: false,
      message: ''
    }
  },
  modal: {
    modal: {
      userDetails: {
        display: false,
        details: {
          name: {
            first: '',
            last: ''
          },
          gender: '',
          email: '',
          phone: '',
          picture: '',
          location: {
            street: {
              number: '',
              street: ''
            },
            city: '',
            postcode: ''
          }
        }
      }
    }
  },
  previousRoute: [],
  navigation: {
    navigationLinks: {
      links: [
        {
          title: NavigationLinkText.item1,
          href: '/app/users',
          className: 'nav-users',
          icon: 'fa-user-circle-o'
        },
        {
          title: NavigationLinkText.item2,
          href: '/app/female-users',
          className: 'nav-female-users',
          icon: 'fa-female'
        },
        {
          title: NavigationLinkText.item3,
          href: '/app/male-users',
          className: 'nav-male-users',
          icon: 'fa-male'
        },
        {
          title: NavigationLinkText.item4,
          href: '/app/products',
          className: 'nav-products',
          icon: 'fa-newspaper-o'
        },
        {
          title: NavigationLinkText.item5,
          href: '/sign-out',
          className: 'nav-sign-out',
          icon: 'fa-sign-out'
        }
      ]
    }
  }
}
