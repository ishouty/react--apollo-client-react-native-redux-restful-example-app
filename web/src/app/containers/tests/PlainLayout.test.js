import PlainLayout from '../PlainLayout'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import 'jsdom-global/register'
import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'

import { initialState } from '../../mocks/initialState'

const mockStore = configureStore([thunk])
const store = mockStore({ ...initialState })

describe('PlainLayout', () => {
  it.only('Should match snapshot and render main within PlainLayout', () => {
    const wrapper = mount(
      <Provider store={store}>
        <PlainLayout children={'hello plain layout'} />
      </Provider>
    )

    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
