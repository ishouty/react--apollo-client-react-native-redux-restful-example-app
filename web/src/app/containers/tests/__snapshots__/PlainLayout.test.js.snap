// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainLayout Should match snapshot and render main within PlainLayout 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PlainLayout>
    <div
      id="layout"
    >
      <div
        id="main"
      >
        <Connect(Banner)>
          <Banner
            banner={
              Object {
                "fail": false,
                "message": "",
                "success": false,
              }
            }
            dispatch={[Function]}
          >
            <div />
          </Banner>
        </Connect(Banner)>
        hello plain layout
      </div>
    </div>
  </PlainLayout>
</Provider>
`;
